# Analyze International Debt Statistics 

Have you ever owed someone money? It could be to an individual, dealership, bank etc. The range can vary from person to person. However, imagine the amount that a whole country may owe ... Let your imagination take over. 

A country does not pick up on debt on purpose, but instead to manage its economy. For example, infrastructure spending is an expensive factor required for a country so that its citizens may live comfortably. 

[The World Bank](https://www.worldbank.org) is the organization that provides debt to countries.

We are going to analyze international debt data collected by The World Bank. We are going to find the answers to the following questions: 

1. What is the number of distinct countries present in the database? 
2. What are the distinct debt indicators?
3. What is the total amount of debt owed by all the countries present in the table, in millions?
4. What country has the highest amount of debt?
5. What is the average amount of debt across different debt indicators?
6. What is the highest amount of principal repayments in the "DT.AMT.DLXF.CD" category?

![image](image.png)


Below is a snapshot of the database:

|country_name|country_code|indicator_name                                                    |indicator_code|debt       |
|------------|------------|------------------------------------------------------------------|--------------|-----------|
|Afghanistan |AFG         |"Disbursements on external debt, long-term (DIS, current US$)"    |DT.DIS.DLXF.CD|72894453.7 |
|Afghanistan |AFG         |"Interest payments on external debt, long-term (INT, current US$)"|DT.INT.DLXF.CD|53239440.1 |
|Afghanistan |AFG         |"PPG, bilateral (AMT, current US$)"                               |DT.AMT.BLAT.CD|61739336.9 |
|Afghanistan |AFG         |"PPG, bilateral (DIS, current US$)"                               |DT.DIS.BLAT.CD|49114729.4 |
|Afghanistan |AFG         |"PPG, bilateral (INT, current US$)"                               |DT.INT.BLAT.CD|39903620.1 |
|Afghanistan |AFG         |"PPG, multilateral (AMT, current US$)"                            |DT.AMT.MLAT.CD|39107845   |
|Afghanistan |AFG         |"PPG, multilateral (DIS, current US$)"                            |DT.DIS.MLAT.CD|23779724.3 |
|Afghanistan |AFG         |"PPG, multilateral (INT, current US$)"                            |DT.INT.MLAT.CD|13335820   |
|Afghanistan |AFG         |"PPG, official creditors (AMT, current US$)"                      |DT.AMT.OFFT.CD|100847181.9|
|Afghanistan |AFG         |"PPG, official creditors (DIS, current US$)"                      |DT.DIS.OFFT.CD|72894453.7 |

Now it is time to execute the SQL queries to answer six questions. 

# What is the number of distinct countries present in the database? 


In [2]:
--num_distinct_countries

SELECT COUNT(DISTINCT country_name) AS total_distinct_countries
FROM international_debt;

total_distinct_countries
0                       124

We can see from the results above that there are a total of 124 countries present in the database. Now we have an idea on what we are working with. 

DISTINCT is a keyword in SQL that eliminates all duplicate records when querying. 

In order to gather the total countries (not being duplicated) we must use the "COUNT()" function. 

It is crucial to create an alias for your query to make it easier to locate and review. The keyword used for an alias is "AS". In this case, we will name our alias "total_distinct_countries". 

The total number of distinct countries will clean up the data to not include duplicate countries. 

Let's move on to the next question. 

# What are the distinct debt indicators? 

In [3]:
--distinct_debt_indicator

SELECT DISTINCT indicator_code AS distinct_debt_indicators
FROM international_debt 
GROUP BY indicator_code
ORDER BY distinct_debt_indicators;

distinct_debt_indicators
0            DT.AMT.BLAT.CD
1            DT.AMT.DLXF.CD
2            DT.AMT.DPNG.CD
3            DT.AMT.MLAT.CD
4            DT.AMT.OFFT.CD
5            DT.AMT.PBND.CD
6            DT.AMT.PCBK.CD
7            DT.AMT.PROP.CD
8            DT.AMT.PRVT.CD
9            DT.DIS.BLAT.CD
10           DT.DIS.DLXF.CD
11           DT.DIS.MLAT.CD
12           DT.DIS.OFFT.CD
13           DT.DIS.PCBK.CD
14           DT.DIS.PROP.CD
15           DT.DIS.PRVT.CD
16           DT.INT.BLAT.CD
17           DT.INT.DLXF.CD
18           DT.INT.DPNG.CD
19           DT.INT.MLAT.CD
20           DT.INT.OFFT.CD
21           DT.INT.PBND.CD
22           DT.INT.PCBK.CD
23           DT.INT.PROP.CD
24           DT.INT.PRVT.CD

We see that there are a total of 25 debt indicators present in the database. This will be useful to know so that we can separate and classify any particular data. 

The GROUP BY clause will group similar debt indicators together. However, since we are using DISTINCT there won't be repetitive debt indicators. 

The ORDER BY clause by default will order our result in ascedning order. 

Let's move on to the next question. 

# What is the total amount of debt owed by all the countries present in the table, in millions?

In [4]:
-- total_debt

SELECT ROUND (SUM(debt) / 1000000, 2) AS total_debt
FROM international_debt;

total_debt
0  3079734.49

The total debt is $3079734.49. 

SUM will obtain the total amount owed by all countries. 

ROUND contains two arugments: value and precision. The precision is the 2 right after the first comma which rounds our solution two decimal places.

Let's move on to the next question.           

# What country has the highest amount of debt?

In [5]:
--highest_debt_country

SELECT country_name, SUM(debt) AS total_debt
FROM international_debt
GROUP BY country_name
ORDER BY total_debt DESC
LIMIT 1;

country_name    total_debt
0        China  2.857935e+11

Here we can see that we selected the column "country_name" 

We then selected DESC which is also known as descending to order the result from biggest to smallest. 

We then used the LIMIT clause to restrict the number of rows to 1 so that we may see the top 1 country who owes the most which was China with a total of $285,793,494,734.2. 

Let's move on to the next question.   

# What is the average amount of debt across different debt indicators?

In [6]:
--avg_debt_per_indicator

SELECT indicator_code AS debt_indicator, indicator_name, AVG(debt) AS average_debt
FROM international_debt
GROUP BY debt_indicator, indicator_name
ORDER BY average_debt DESC
LIMIT 10;

debt_indicator  ...  average_debt
0  DT.AMT.DLXF.CD  ...  5.904868e+09
1  DT.AMT.DPNG.CD  ...  5.161194e+09
2  DT.DIS.DLXF.CD  ...  2.152041e+09
3  DT.DIS.OFFT.CD  ...  1.958983e+09
4  DT.AMT.PRVT.CD  ...  1.803694e+09
5  DT.INT.DLXF.CD  ...  1.644024e+09
6  DT.DIS.BLAT.CD  ...  1.223139e+09
7  DT.INT.DPNG.CD  ...  1.220411e+09
8  DT.AMT.OFFT.CD  ...  1.191188e+09
9  DT.AMT.PBND.CD  ...  1.082624e+09

[10 rows x 3 columns]

The debt indicator displayed to the left of the table allows us to summarize and review the following output. 

LIMIT set at 10 provides a range of the top 10 owing debt indicators. 

ORDER BY arranged the average debt in descedning order so that the top owing debt is displayed first. In this case is Principal repayments on external debt, long-term has a debt of $5904868401.499193. 

# What is the highest amount of principal repayments in the "DT.AMT.DLXF.CD" category?

In [9]:
--highest_principal_repayment

SELECT country_name, indicator_name
FROM international_debt 
WHERE indicator_code = 'DT.AMT.DLXF.CD' AND debt = (SELECT MAX(debt) 
FROM international_debt);

country_name                                     indicator_name
0        China  Principal repayments on external debt, long-te...